core/node/rpc/add_event.go

```diff
@@ -77,11 +77,11 @@ func (s *Service) addParsedEvent(
 		return err
 	}
 
-	if len(chainAuthArgsList) > 0 {
+	if len(verifications.OneOfChainAuths) > 0 {
 		isEntitled := false
 		var err error
 		// Determine if any chainAuthArgs grant entitlement
-		for _, chainAuthArgs := range chainAuthArgsList {
+		for _, chainAuthArgs := range verifications.OneOfChainAuths {
 			isEntitled, err = s.chainAuth.IsEntitled(ctx, s.config, chainAuthArgs)
 			if err != nil {
```

Comment:
we want to return errors so that clients can retry - but since it's a oneOf, maybe one succeeds and one errors, in which case we could just move forward? It's a good note.